### ISSUE TITLE
Deleting snapshot causes NPE with no useful error information

### ISSUE THREAD

**Issue by bradvido** [elastic/elasticsearch#15450]
I create nightly snapshots of '_all' indices. They are created in a file-system repository on a Windows Server 2012 shared folder. The snapshots are created successfully and I can successfully restore data from them, and *some* can be successfully deleted.

However, when I try to some particular snapshots, ES return a 500 status code and only this info:
`{"error":"NullPointerException[null]","status":500}`
Since there is no stacktrace and no other info (I checked ES node logs of the cluster), I have no idea how to fix it.

This is on ES 1.7.3.

I don't think it is permissions issues since only some of the snapshots fail to be deleted. Others have been deleted successfully from the same repo.

The bottom line is that it would be nice to get more detailed error info on the failure.

**Comment by clintongormley** [comment]
@imotov @ywelsch any ideas how to figure this out?

**Comment by s1monw** [comment]
this is very very likely fixed in 2.0 and above - there are tons of places where we just don't do the right thing with exceptions in 1.7.3. @bradvido is there any chance you can try reproducing this with 2.1?

**Comment by bradvido** [comment]
Unfortunately, we don't have any plans to move to 2.0+ soon.  This is still occurring though. 
Is there any way to force ES to print the stacktrace?

**Comment by imotov** [comment]
@bradvido when you said 
> However, when I try to some particular snapshots

what did you mean? What are you trying to do with these snapshots?

**Comment by bradvido** [comment]
@imotov I am trying to delete them. Some snapshots can be deleted successfully, others can not

**Comment by imotov** [comment]
@bradvido could you email me `snapshot-XXXXXX` and `metadata-XXXXXX` files from the root directory of your repository, where `XXXXXXX` is the name of the snapshot that you cannot delete?

**Comment by clintongormley** [comment]
This issue is fairly old and there hasn't been much activity on it. Closing, but please re-open if it still occurs.
